pkg/tsdb/elasticsearch/time_series_query.go

```diff
@@ -78,18 +78,60 @@ func (e *timeSeriesQuery) processQuery(q *Query, ms *es.MultiSearchRequestBuilde
 	}
 
 	if len(q.BucketAggs) == 0 {
-		if len(q.Metrics) == 0 || !(q.Metrics[0].Type == "raw_document" || q.Metrics[0].Type == "raw_data") {
+		// If no metrics, then the query is invalid
+		if len(q.Metrics) == 0 {
 			result.Responses[q.RefID] = backend.DataResponse{
 				Error: fmt.Errorf("invalid query, missing metrics and aggregations"),
 			}
 			return nil
 		}
+
 		metric := q.Metrics[0]
-		b.Size(metric.Settings.Get("size").MustInt(500))
-		b.SortDesc(e.client.GetTimeField(), "boolean")
-		b.SortDesc("_doc", "")
-		b.AddDocValueField(e.client.GetTimeField())
-		return nil
+		isLogs := metric.Type == "logs"
+
+		// Let's group these into the same variable, as the resulting query is the same
+		isRawData := metric.Type == "raw_data" || metric.Type == "raw_document"
+
+		// Only raw data and logs queries can be without aggregations
+		if !(isRawData || isLogs) {
+			result.Responses[q.RefID] = backend.DataResponse{
+				Error: fmt.Errorf("invalid query, missing metrics and aggregations"),
+			}
+			return nil
+		}
+
+		if isRawData {
+			b.Size(metric.Settings.Get("size").MustInt(500))
+			b.SortDesc(e.client.GetTimeField(), "boolean")
+			b.SortDesc("_doc", "")
+			b.AddDocValueField(e.client.GetTimeField())
+			return nil
+		}
+
+		if isLogs {
```

Comment:
In the future, we should refactor this to separate functions, but I want to do minimal changes. When everything is passing, then refactor can be done in separate PR.